src/coreclr/tools/Common/Compiler/GenericCycleDetection/ModuleCycleInfo.cs

```diff
@@ -142,7 +142,20 @@ private bool IsDeepPossiblyCyclicInstantiation(TypeDesc type, ref int breadthCou
                 {
                     case TypeFlags.Array:
                     case TypeFlags.SzArray:
-                        return IsDeepPossiblyCyclicInstantiation(((ParameterizedType)type).ParameterType, ref breadthCounter, seenTypes);
+                        TypeDesc parameterType = type;
+                        int arrayNesting = 0;
+                        do
+                        {
+                            parameterType = ((ParameterizedType)parameterType).ParameterType;
+                            arrayNesting++;
+                        } while (parameterType.IsArray);
```

Comment:
Should the the _depthCutoff condition checked in the loop for early bail?

Reply:
> Should the _depthCutoff condition checked in loop for early bail? In case of cycles, it may run into infinite recursion.

You mean whether we can have array A, whose element is B, which is another array with element A? That is impossible to construct in the managed type system.